### ISSUE TITLE
Segmentation fault in MILP computation

### ISSUE THREAD

**Issue by ThomasBreuer** [oscar-system/Polymake.jl#144]
The following Polymake input follows [some examples in polymake docs](https://www.polymake.org/release_docs/latest/polytope.html)
```
$p = new Polytope( INEQUALITIES => [ [1,1,-1],[-1,0,1],[7,-1,-1] ] );
$obj = new Vector( [0,-1,-1] );
$intvar = new Set<Int>( [0,1,2] );
$milp = $p -> MILP( LINEAR_OBJECTIVE => $obj,
                    INTEGER_VARIABLES => $intvar );
$result = $milp -> MINIMAL_VALUE;
print $result;
```
and yields -7.
Using the syntax translation rules from [the README of Polymake.jl](https://github.com/oscar-system/Polymake.jl),
I tried to do the same with Polymake.jl, as follows.
```
using Polymake
p = @pm Polytope.Polytope( :INEQUALITIES => [1 1 -1; -1 0 1; 7 -1 -1] )
obj = [0,-1,-1]
intvar = Set{Int}( [0,1,2] )
milp = p.MILP( :LINEAR_OBJECTIVE => obj,
               :INTEGER_VARIABLES => intvar )
```
Here I get the following error message.
```
polymake:  WARNING: available properties insufficient to compute 'MILP'
ERROR: Exception occured at Polymake side:
property MILP not created as expected at ...
```
With help from Sebastian Gutsche, I tried the following variant.
```
p = @pm Polytope.Polytope( :INEQUALITIES => [1 1 -1; -1 0 1; 7 -1 -1] )
lp = @pm Polytope.MixedIntegerLinearProgram(
    :LINEAR_OBJECTIVE => [0,-1,-1],
    :INTEGER_VARIABLES => [0,1,2] )
p.MILP = lp
p.MILP.MINIMAL_VALUE
```
Here I get a segmentation fault.

**Comment by kalmarek** [comment]
tosimplex uses omp threads which for some reason (we're investigating right now) clashes with julia gc (?!);

as a workaround you can either `OMP_NUM_THREADS=1 julia`, or `ENV["OMP_NUM_THREADS"] = 1` **before** `using Polymake`

```julia
julia> ENV["OMP_NUM_THREADS"] = 1
1

julia> using Polymake
polymake version 3.4
Copyright (c) 1997-2019
Ewgenij Gawrilow, Michael Joswig (TU Berlin)
https://polymake.org

This is free software licensed under GPL; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


julia> p = @pm Polytope.Polytope( :INEQUALITIES => [1 1 -1; -1 0 1; 7 -1 -1] )
type: Polytope<Rational>

INEQUALITIES
1 1 -1
-1 0 1
7 -1 -1
1 0 0



julia> obj = [0,-1,-1]
3-element Array{Int64,1}:
  0
 -1
 -1

julia> intvar = [0,1,2]
3-element Array{Int64,1}:
 0
 1
 2

julia> lp = @pm Polytope.MixedIntegerLinearProgram( LINEAR_OBJECTIVE = obj, INTEGER_VARIABLES = intvar)
type: MixedIntegerLinearProgram<Rational>

LINEAR_OBJECTIVE
0 -1 -1

INTEGER_VARIABLES
{0 1 2}


julia> p.MILP = lp
type: MixedIntegerLinearProgram<Rational>

LINEAR_OBJECTIVE
0 -1 -1

INTEGER_VARIABLES
{0 1 2}


julia> p.MILP.MINIMAL_VALUE
polymake: used package tosimplex
  Dual simplex algorithm implemented by Thomas Opfer

-7

```

**Comment by ThomasBreuer** [comment]
Thanks a lot for this quick workaround.
Now it works, even in situations where the polytope does not contain lattice points.

**Comment by kalmarek** [comment]
a patch for this has been written in June 2018:
https://github.com/JuliaLang/julia/pull/27020
but has not been merged

**Comment by kalmarek** [comment]
a new pull already approved is pending:
https://github.com/JuliaLang/julia/pull/33284

**Comment by saschatimme** [comment]
This is resolved with Julia 1.3
